server/src/game-implementations/games/ddr.ts

```diff
@@ -0,0 +1,332 @@
+				case flarePoints < 7000:
+					return "VENUS";
+				case flarePoints < 8000:
+					return "VENUS+";
+				case flarePoints < 9000:
+					return "VENUS++";
+				case flarePoints < 10000:
+					return "VENUS+++";
+				case flarePoints < 11500:
+					return "EARTH";
+				case flarePoints < 13000:
+					return "EARTH+";
+				case flarePoints < 14500:
+					return "EARTH++";
+				case flarePoints < 16000:
+					return "EARTH+++";
+				case flarePoints < 18000:
+					return "MARS";
+				case flarePoints < 20000:
+					return "MARS+";
+				case flarePoints < 22000:
+					return "MARS++";
+				case flarePoints < 24000:
+					return "MARS+++";
+				case flarePoints < 26500:
+					return "JUPITER";
+				case flarePoints < 29000:
+					return "JUPITER+";
+				case flarePoints < 31500:
+					return "JUPITER++";
+				case flarePoints < 34000:
+					return "JUPITER+++";
+				case flarePoints < 36750:
+					return "SATURN";
+				case flarePoints < 39500:
+					return "SATURN+";
+				case flarePoints < 42250:
+					return "SATURN++";
+				case flarePoints < 45000:
+					return "SATURN+++";
+				case flarePoints < 48750:
+					return "URANUS";
+				case flarePoints < 52500:
+					return "URANUS+";
+				case flarePoints < 56250:
+					return "URANUS++";
+				case flarePoints < 60000:
+					return "URANUS+++";
+				case flarePoints < 63750:
+					return "NEPTUNE";
+				case flarePoints < 67500:
+					return "NEPTUNE+";
+				case flarePoints < 71250:
+					return "NEPTUNE++";
+				case flarePoints < 75000:
+					return "NEPTUNE+++";
+				case flarePoints < 78750:
+					return "SUN";
+				case flarePoints < 82500:
+					return "SUN+";
+				case flarePoints < 86250:
+					return "SUN++";
+				case flarePoints < 90000:
+					return "SUN+++";
+			}
+
+			return "WORLD";
+		},
+	},
+	defaultMergeRefName: "Best Score",
+	derivers: {
+		grade: ({ score, lamp }) => {
+			if (lamp === "FAILED") {
+				return "E";
+			}
+
+			return GetGrade(DDR_GBOUNDARIES, score);
+		},
+		percent: ({ score }) => {
+			return score / 1_000_000;
+		},
+	},
+	goalCriteriaFormatters: DDR_GOAL_FMT,
+	goalProgressFormatters: DDR_GOAL_PG_FMT,
+	goalOutOfFormatters: DDR_GOAL_OO_FMT,
+	pbMergeFunctions: [],
+	profileCalcs: {
+		flareSkill: async (game: Game, playtype: Playtype, userID: integer) => {
+			const sc: Array<PBScoreDocumentWithSong> = await db["personal-bests"].aggregate([
+				{
+					$match: {
+						userID,
+						game,
+						playtype,
+						isPrimary: true,
+						[`calculatedData.flareSkill`]: { $type: "number" },
+					},
+				},
+				{
+					$lookup: {
+						from: "songs-ddr",
+						localField: "songID",
+						foreignField: "id",
+						as: "song",
+					},
+				},
+				{
+					$unwind: {
+						path: "$song",
+					},
+				},
+				{
+					$sort: {
+						[`calculatedData.flareSkill`]: -1,
+					},
+				},
+			]);
+
+			if (sc.length === 0) {
+				return null;
+			}
+
+			let classicIndex = 0;
+			let goldIndex = 0;
+			let whiteIndex = 0;
+
+			for (const score of sc) {
```

Comment:
whoever came up with this rating system is insane. We might not even want to bother implementing this.

Reply:
That is to say: this isn't even really measuring skill or anything, it's just a whole bunch of profile noise.

Reply:
This ranking system they added to DDR World is clearly made as an incentive to use the new Flare system they added at the end of A3. Personally, I still find it interesting. Maybe some other custom ranking system could be implemented too but I'm not the one who'll do that.

By the way, out of curiousity, could several ranking systems cohabitate on Tachi ?

Reply:
Thought : if you prefer something more universal than just blocked behind the Flare accessibility, we could potentially keep the same rating system but with score equivalences to Flare level recommendations. Something along those lines :


![image](https://github.com/user-attachments/assets/52ba250a-c72c-481a-b113-1f89d91fecb8)